test/sslapitest.c

```diff
@@ -737,83 +609,50 @@ static int test_tlsext_status_type(void)
     BIO *certbio = NULL;
 
     if (!create_ssl_ctx_pair(TLS_server_method(), TLS_client_method(), &sctx,
-                             &cctx, cert, privkey)) {
-        printf("Unable to create SSL_CTX pair\n");
+                             &cctx, cert, privkey))
         return 0;
-    }
 
-    if (SSL_CTX_get_tlsext_status_type(cctx) != -1) {
-        printf("Unexpected initial value for "
-               "SSL_CTX_get_tlsext_status_type()\n");
+    if (SSL_CTX_get_tlsext_status_type(cctx) != -1)
         goto end;
-    }
 
     /* First just do various checks getting and setting tlsext_status_type */
 
     clientssl = SSL_new(cctx);
-    if (SSL_get_tlsext_status_type(clientssl) != -1) {
-        printf("Unexpected initial value for SSL_get_tlsext_status_type()\n");
-        goto end;
-    }
-
-    if (!SSL_set_tlsext_status_type(clientssl, TLSEXT_STATUSTYPE_ocsp)) {
-        printf("Unexpected fail for SSL_set_tlsext_status_type()\n");
+    if (!TEST_int_eq(SSL_get_tlsext_status_type(clientssl), -1)
+            || !TEST_true(SSL_set_tlsext_status_type(clientssl,
+                                                      TLSEXT_STATUSTYPE_ocsp))
+            || !TEST_int_eq(SSL_get_tlsext_status_type(clientssl),
+                            TLSEXT_STATUSTYPE_ocsp))
         goto end;
-    }
-
-    if (SSL_get_tlsext_status_type(clientssl) != TLSEXT_STATUSTYPE_ocsp) {
-        printf("Unexpected result for SSL_get_tlsext_status_type()\n");
-        goto end;
-    }
 
     SSL_free(clientssl);
     clientssl = NULL;
 
-    if (!SSL_CTX_set_tlsext_status_type(cctx, TLSEXT_STATUSTYPE_ocsp)) {
-        printf("Unexpected fail for SSL_CTX_set_tlsext_status_type()\n");
-        goto end;
-    }
-
-    if (SSL_CTX_get_tlsext_status_type(cctx) != TLSEXT_STATUSTYPE_ocsp) {
-        printf("Unexpected result for SSL_CTX_get_tlsext_status_type()\n");
+    if (!SSL_CTX_set_tlsext_status_type(cctx, TLSEXT_STATUSTYPE_ocsp)
+     || SSL_CTX_get_tlsext_status_type(cctx) != TLSEXT_STATUSTYPE_ocsp)
         goto end;
-    }
 
     clientssl = SSL_new(cctx);
-
-    if (SSL_get_tlsext_status_type(clientssl) != TLSEXT_STATUSTYPE_ocsp) {
-        printf("Unexpected result for SSL_get_tlsext_status_type() (test 2)\n");
+    if (SSL_get_tlsext_status_type(clientssl) != TLSEXT_STATUSTYPE_ocsp)
         goto end;
-    }
-
     SSL_free(clientssl);
     clientssl = NULL;
 
     /*
      * Now actually do a handshake and check OCSP information is exchanged and
      * the callbacks get called
      */
-
```

Comment:
[marker] I need to take a pause here, will get back to it later, when my eyes have uncrossed...